app/services/validation_beta_service.rb

```diff
@@ -110,14 +110,23 @@ def tell_induction_coordinators_who_are_mentors_to_add_validation_information
   end
 
   def tell_induction_coordinators_we_asked_ects_and_mentors_for_information
-    InductionCoordinatorProfile.find_each do |ic|
-      ic.schools.not_opted_out.each do |school|
-        if chosen_programme_and_not_in_beta(school)
-          send_induction_coordinators_we_asked_ects_and_mentors_for_information(ic, school)
-          break
+    school_cohorts = SchoolCohort
+      .where(
+        id: ParticipantProfile::ECF.where("created_at > ?", Date.new(2021, 9, 7)).select(:school_cohort_id),
+        induction_programme_choice: %i[core_induction_programme full_induction_programme],
+      )
+
+    InductionCoordinatorProfile
+      .includes(:schools)
+      .where(schools: { id: school_cohorts.select(:school_id) })
+      .find_each do |ic|
+        ic.schools.not_opted_out.each do |school|
+          if chosen_programme_and_not_in_beta(school)
+            send_induction_coordinators_we_asked_ects_and_mentors_for_information(ic, school)
+            break
+          end
```

Comment:
If an IC has two schools, and added participants to one of them, we would surely want to send an email with that school name.

But the code here says 'take that IC, for each of their schools check if they are not opted out, check they have chosen programme, and not in beta and send them email with that school'. 

Which I guess will give us the wrong school name a bunch of times.

Not sure if it matters much, but if it does then we should probably have the `each` iteration on schools rather than on ICs - and maybe store a list of ICs that already saw an email.

Reply:
Good catch! Fixed.